internal/services/network/network_interface_application_gateway_association_resource_test.go

```diff
@@ -173,6 +174,9 @@ func (NetworkInterfaceApplicationGatewayBackendAddressPoolAssociationResource) d
 	}
 	config.InterfaceIPConfigurationPropertiesFormat.ApplicationGatewayBackendAddressPools = &updatedPools
 
+	// renew timeout value of ctx or it can be timeout
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*30)
```

Comment:
this is only fixing the test? we should be fixing the resource so its not required here?

Reply:
the timeout value for the resource is enough in my opinion, because the `ctx` variable was passed by AccTest Framework and the timeout vaule was set there in the first Create, and timeout is more easier to meet here in `data.CheckWithClient` for it is one timeout value for two Create.

Reply:
Hi @wuxu92 - This won't help unfortunately, the wrapping context deadline will still cancel the operation. The points at which this is failing there is clearly a problem with the service, and the deadline should not be extended until the source of the failure is known.  Can you discuss with the relevant service team on your side to look at the intermittent failure?

Reply:
you are right @jackofallops there are a bunch of failures cases in this run. it should be a server-side problem and I am closing this PR for now.